### ISSUE TITLE
Escaped single quotes break the parser

### ISSUE THREAD

**Issue by Ivan-Johnson** [GothenburgBitFactory/taskwarrior#2189]
The database is corrupted after making a task that contain a backslash that is immediately followed by a single quote. I am using task warrior version 2.5.1; the complete output of `task diag` is included at the end. `task add "foo \' bar"` and `task 1 append "foo \' bar"` both cause the database to be corrupted when executed in bash (to be clear, the arguments that bash actually passes to task warrior contains both the backslash and the single quote, but not the surrounding double quotes).

Immediately after running either of the commands listed above I am able to run `task list` a single time, but subsequent executions of `task list` fail with the message "Unrecognized Taskwarrior file format or blank line in data. in /home/i/.task/completed.data at line 1". The line that this message references is `[]`. After removing that line `task list` functions normally, despite the fact that the last line in the backlog.data file, in its entirety, is `{"description":"26707 foo \\`.

By the way, I was originally getting the same error message that is described in #2018:

> task: /build/task/src/task-2.5.1/src/Task.cpp:1955: float Task::urgency_age() const: Assertion `has ("entry")' failed.

It appears as though this message only appears when you append/modify a task that isn't the most recently created task. In this case, `task list` fails the very first time it's executed.

Here is the complete output of `task diag`:

> 
> task 2.5.1
>    Platform: Linux
> 
> Compiler
>     Version: 8.2.1 20180831
>        Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
>  Compliance: C++11
> 
> Build Features
>       Built: Nov  9 2018 23:33:36
>       CMake: 3.12.4
>     libuuid: libuuid + uuid_unparse_lower
>   libgnutls: 3.5.19
>  Build type: 
> 
> Configuration
>        File: /home/i/.taskrc (found), 1406 bytes, mode 100640
>        Data: /home/i/.task (found), dir, mode 40750
>     Locking: Enabled
>          GC: Enabled
>     $VISUAL: emacs
>      Server: 
>       Trust: strict
> Certificate: , not readable, 0 bytes
>         Key: , not readable, 0 bytes
>     Ciphers: NORMAL
>       Creds: 
> 
> Hooks
>      System: Enabled
>    Location: /home/i/.task/hooks
>              (-none-)
> 
> Tests
>       $TERM: rxvt-unicode-256color (120x30)
>        Dups: Scanned 522 tasks for duplicate UUIDs:
>              No duplicates found
>  Broken ref: Scanned 522 tasks for broken references:
>              No broken references found
> 
>

**Comment by JohnsonIvan** [comment]
I just noticed that this exact bug is described in [a comment](https://github.com/GothenburgBitFactory/taskwarrior/issues/1960#issuecomment-399706005) on a closely related issue (#1960).

**Comment by Mooie** [comment]
+1

I'm getting the error "Unrecognized Taskwarrior file format or blank line in data. in /Users/home-dir/.task/completed.data at line 1" after adding a task containing an escaped single quote. 

"task undo" generates  "Unknown error. Please report."

Deleting the first line of "completed.data" resolved the issue.  Although, the task that I originally added has disappeared.

**Comment by tbabej** [comment]
This problem is somewhat mitigated on newer releases and on the development branch, but not completely.

On 2.5.3 the task (silently!) fails to be added:
```
[warrior@tw-v2.5.3 ~]$ task add test
Created task 1.
[warrior@tw-v2.5.3 ~]$ task add "foo \' bar"
Created task 2.
[warrior@tw-v2.5.3 ~]$ task list

ID Age  Description Urg 
 1 5s   test           0

1 task
[warrior@tw-v2.5.3 ~]$ cat ~/.task/pending.data 
[description:"test" entry:"1622027936" modified:"1622027936" status:"pending" uuid:"5a0fd5ea-0406-4fcc-9d44-6c74823c56ad"]
```

On the 2.6.0 development branch, the task is added, but everything after the escaped quote is 
```
[warrior@tw-2.6.0 ~]$ task add test
Created task 1.
[warrior@tw-2.6.0 ~]$ task add "foo \' bar"
Created task 2.
[warrior@tw-2.6.0 ~]$ task list

ID Age  Description Urg 
 1 5s   test           0
 2 1s   foo \          0

2 tasks
Filter: ( status = pending )
[warrior@tw-2.6.0 ~]$ cat ~/.task/pending.data 
[description:"test" entry:"1622028001" modified:"1622028001" status:"pending" uuid:"0ba4ae49-a0ed-4ff6-8cd9-f7ae7a39f899"]
[description:"foo \\" entry:"1622028005" modified:"1622028005" status:"pending" uuid:"3bcd76d6-cccb-4591-80ae-11ece6440fd8"]
```

As a sidenote, a lot of bugs were fixed between 2.5.1 and 2.5.3 (albeit, unfortunately, not this one :slightly_frowning_face: ). I'd recommend upgrading.
